Doc/library/asyncio-eventloop.rst

```diff
@@ -1206,7 +1206,13 @@ Executing code in thread or process pools
                   pool, cpu_bound)
               print('custom process pool', result)
 
-      asyncio.run(main())
+      if __name__ == '__main__':
+          asyncio.run(main())
+
+   Note that on Windows and macOS, the entry point guard may be necessary as the
+   example would otherwise cause a :exc:`RuntimeError`. For more details, see the
+   `Safe importing of main module
+   <https://docs.python.org/3/library/multiprocessing.html#the-spawn-and-forkserver-start-methods>`_.
```

Comment:
I propose to simplify the note to the following:

Note that the entry point guard (`if __name__ == '__main__'`) is required for option 3, due to the peculiarities of `multiprocessing`, which is used by `ProcessPoolExecutor`. See (link).

There should be three cross-references:
- multiprocessing (the module)
- ProcessPoolExecutor (the class)
- Safe importing of main module
For the latter I recommend to add a label to the target section in multiprocessing.rst rather than linking to the website. (@CAM-Gerlach can provide details on how to do this.)

Reply:
My understanding though is that option 3 fails when `spawn`/`forkserver` is the process start method, and only Windows + mac default to `spawn`, while unix defaults to `fork`, so saying 'is required' may be a bit too strong. If 'may be required' instead, then the reader might wonder why 'may', in which case I think a link to https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods somewhere would help explain things.

Reply:
That's too subtle for the readers. We're not stating the law, we're just explaining the example.

Reply:
See [my comment](https://github.com/python/cpython/pull/93457/files#r996388391) for a suggestion implementing the above and the details Guido mentioned.